### PR TITLE
[CORE] Move i8, u8 Multiply and Divide ops into template

### PR DESCRIPTION
**Pull request by barnasm1**
### Details:
 - *Move i8, u8 Multiply and Divide ops into template*

### Tickets:
 - *[CVS-136165](https://jira.devtools.intel.com/browse/CVS-136165)*

### CODE REVIEW

**Review by praasz — CHANGES_REQUESTED** [review]

File: src/plugins/template/backend/ops/divide.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    case ov::element::i8:
+        return evaluate<ov::element::i8>(ov::as_type_ptr<ov::op::v1::Divide>(node), outputs, inputs);
+    case ov::element::i16:
+        return evaluate<ov::element::i16>(ov::as_type_ptr<ov::op::v1::Divide>(node), outputs, inputs);
+    case ov::element::u8:
+        return evaluate<ov::element::u8>(ov::as_type_ptr<ov::op::v1::Divide>(node), outputs, inputs);
+    case ov::element::u16:
+        return evaluate<ov::element::u16>(ov::as_type_ptr<ov::op::v1::Divide>(node), outputs, inputs);
+    default:
+        OPENVINO_THROW("Unhandled data type ", node->get_element_type().get_type_name(), " in evaluate_node()");
```
[the remark this answers is not in the document]
Comment (barnasm1) [reply src/plugins/template/backend/ops/divide.cpp]: Done

File: src/plugins/template/tests/functional/op_reference/multiply.cpp [hunk 2]
```diff
@@ -115,15 +106,79 @@ std::vector<MultiplyParams> generateParamsForMultiplyFloat() {
...
+        std::vector<T>{1, 2, 3, 4, 6, 1},
+        std::vector<T>{12, 24, 36,  48,  72,  12, 24, 48,  72,  96,  144, 24, 36, 72,  108, 144, 216, 36,
+                       48, 96, 144, 192, 288, 48, 60, 120, 180, 240, 360, 60, 72, 144, 216, 288, 432, 72})};
+
+    return params;
+}
+
+template <>
+std::vector<MultiplyParams> generateParamsForMultiply<element::Type_t::i8>() {
+    constexpr element::Type_t IN_ET = element::Type_t::i8;
```
[the remark this answers is not in the document]
Comment (barnasm1) [reply src/plugins/template/tests/functional/op_reference/multiply.cpp]: Done

File: src/plugins/template/backend/ops/divide.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+#include "openvino/reference/divide.hpp"
+
+#include "evaluate_node.hpp"
+
+template <ov::element::Type_t ET>
+bool evaluate(const std::shared_ptr<ov::op::v1::Divide>& op,
+              ov::TensorVector& outputs,
+              const ov::TensorVector& inputs) {
+    using T = typename ov::element_type_traits<ET>::value_type;
```
[the remark this answers is not in the document]
Comment (barnasm1) [reply src/plugins/template/backend/ops/divide.cpp]: Done

File: src/plugins/template/tests/functional/op_reference/multiply.cpp [hunk 4]
```diff
@@ -115,15 +106,79 @@ std::vector<MultiplyParams> generateParamsForMultiplyFloat() {
...
+        ov::PartialShape{3, 2, 1},
+        ov::PartialShape{1, 6},
+        IN_ET,
+        std::vector<T>{12, 24, 36, 48, 60, 72},
+        std::vector<T>{1, 2, 3, 4, 6, 1},
+        std::vector<T>{12, 24, 36,  48,  72, 12, 24, 48,  72,  96,  144, 24, 36, 72,  108, 144, 216, 36,
+                       48, 96, 144, 192, 32, 48, 60, 120, 180, 240, 104, 60, 72, 144, 216, 32,  176, 72})};
+
+    return params;
+}
```
Comment (t-jankowski) [inline src/plugins/template/tests/functional/op_reference/multiply.cpp:157]: Unless OV specification declares certain behavior for overflow, it shouldn't be tested.

File: src/plugins/template/tests/functional/op_reference/multiply.cpp [hunk 5]
```diff
@@ -115,15 +106,79 @@ std::vector<MultiplyParams> generateParamsForMultiplyFloat() {
...
+        ov::PartialShape{3, 2, 1},
+        ov::PartialShape{1, 6},
+        IN_ET,
+        std::vector<T>{12, 24, 36, 48, 60, 72},
+        std::vector<T>{1, 2, 3, 4, 6, 1},
+        std::vector<T>{12, 24, 36,   48,  72, 12, 24, 48,  72,  96,  -112, 24, 36, 72,   108, -112, -40, 36,
+                       48, 96, -112, -64, 32, 48, 60, 120, -76, -16, 104,  60, 72, -112, -40, 32,   -80, 72})};
+
+    return params;
+}
```
Comment (t-jankowski) [inline src/plugins/template/tests/functional/op_reference/multiply.cpp:140]: Testing overflowed for `signed` is risky - it's undefined.

File: src/plugins/template/tests/functional/op_reference/multiply.cpp (hunk 4 above)
Comment (barnasm1) [reply src/plugins/template/tests/functional/op_reference/multiply.cpp]: Dixed

File: src/plugins/template/tests/functional/op_reference/multiply.cpp (hunk 5 above)
Comment (barnasm1) [reply src/plugins/template/tests/functional/op_reference/multiply.cpp]: Fixed

**Review by t-jankowski — APPROVED** [review]

**Review by praasz — APPROVED** [review]

**Pull request merged**